From 655f3b24c05e6bd5aba8a8b208743a6290e46fef Mon Sep 17 00:00:00 2001
Message-Id: <655f3b24c05e6bd5aba8a8b208743a6290e46fef.1422661123.git.feitong.yi@intel.com>
From: Hong Liu <hong.liu@intel.com>
Date: Thu, 27 Nov 2014 13:18:18 +0800
Subject: [PATCH 1/4] FOR_UPSTREAM [VPG]: drm/i915: init hdmi audio before
 init connector

hdmi connector init code will call intel_hdmi_detect(), which tries to
schedule audio_wq that is not inited at that time.

Issue: IMINAN-14752
Change-Id: I0a4954707853e03416dafeaf9ef90d4af5e44c0e
Signed-off-by: Hong Liu <hong.liu@intel.com>
---
 drivers/gpu/drm/i915/intel_hdmi.c |    9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_hdmi.c b/drivers/gpu/drm/i915/intel_hdmi.c
index 05f93f3..3fe5a79 100644
--- a/drivers/gpu/drm/i915/intel_hdmi.c
+++ b/drivers/gpu/drm/i915/intel_hdmi.c
@@ -1285,8 +1285,8 @@ intel_hdmi_detect(struct drm_connector *connector, bool force)
 		}
 #endif
 	}
-	if (&dev_priv->hdmi_audio_wq)
-		schedule_work(&dev_priv->hdmi_audio_wq);
+
+	schedule_work(&dev_priv->hdmi_audio_wq);
 
 #ifdef CONFIG_EXTCON
 	if (strlen(intel_connector->hotplug_switch.name) != 0) {
@@ -2048,9 +2048,6 @@ void intel_hdmi_init(struct drm_device *dev, int hdmi_reg, enum port port)
 	intel_dig_port->hdmi.hdmi_reg = hdmi_reg;
 	intel_dig_port->dp.output_reg = 0;
 
-	intel_hdmi_init_connector(intel_dig_port, intel_connector);
-	intel_connector->panel.fitting_mode = 0;
-
 #ifdef CONFIG_SUPPORT_LPDMA_HDMI_AUDIO
 	INIT_WORK(&dev_priv->hdmi_audio_wq, i915_had_wq);
 	hdmi_priv = kzalloc(sizeof(struct hdmi_audio_priv), GFP_KERNEL);
@@ -2071,4 +2068,6 @@ void intel_hdmi_init(struct drm_device *dev, int hdmi_reg, enum port port)
 	}
 #endif
 
+	intel_hdmi_init_connector(intel_dig_port, intel_connector);
+	intel_connector->panel.fitting_mode = 0;
 }
-- 
1.7.9.5

